src/cascadia/TerminalSettingsModel/CascadiaSettings.cpp

```diff
@@ -265,15 +265,34 @@ winrt::Microsoft::Terminal::Settings::Model::Profile CascadiaSettings::Duplicate
 {
     THROW_HR_IF_NULL(E_INVALIDARG, source);
 
-    auto duplicated{ _userDefaultProfileSettings->CreateChild() };
+    winrt::com_ptr<Profile> duplicated;
+    if (_userDefaultProfileSettings)
+    {
+        duplicated = _userDefaultProfileSettings->CreateChild();
+    }
+    else
+    {
+        duplicated = winrt::make_self<Profile>();
+    }
     _allProfiles.Append(*duplicated);
 
     if (!source.Hidden())
     {
         _activeProfiles.Append(*duplicated);
     }
 
-    const winrt::hstring newName{ fmt::format(L"{} ({})", source.Name(), RS_(L"CopySuffix")) };
+    winrt::hstring newName{ fmt::format(L"{} ({})", source.Name(), RS_(L"CopySuffix")) };
+
+    // Check if this name already exists and if so, append a number
+    for (uint32_t candidateIndex = 0; candidateIndex < _allProfiles.Size() + 1; ++candidateIndex)
+    {
+        if (std::none_of(begin(_allProfiles), end(_allProfiles), [&](auto&& profile) { return profile.Name() == newName; }))
+        {
+            break;
+        }
+        // There is a theoretical unsigned integer wraparound, which is OK
+        newName = fmt::format(L"{} ({}{})", source.Name(), RS_(L"CopySuffix"), candidateIndex);

```

Comment:
Won't this create `Blah (Copy0)`? That seems somewhat unergonomic.

You may want to use candidateIndex+2 (so that we skip "copy 1") and split the number and the copy suffix out with a space, like `Blah (Copy 2)`